src/main/app.ts

```diff
@@ -43,6 +53,170 @@ interface IClearHistoryOptions {
   userSetPythonEnvs: boolean;
 }
 
+const minimumWindowSpacing = 15;
+const windowSpacing = 30;
+
+class SessionWindowManager implements IDisposable {
+  constructor(options: SessionWindowManager.IOptions) {
+    this._options = options;
+  }
+
+  createNewEmptyWindow(): SessionWindow {
+    return this.createNew(ContentViewType.Welcome);
+  }
+
+  restoreLabWindow(sessionConfig?: SessionConfig): SessionWindow {
+    return this.createNew(ContentViewType.Lab, sessionConfig, true);
+  }
+
+  createNewLabWindow(sessionConfig?: SessionConfig): SessionWindow {
+    return this.createNew(ContentViewType.Lab, sessionConfig);
+  }
+
+  getOrCreateEmptyWindow(): SessionWindow {
+    const emptySessionWindow = this._windows.find(sessionWindow => {
+      return sessionWindow.contentViewType === ContentViewType.Welcome;
+    });
+
+    if (emptySessionWindow) {
+      return emptySessionWindow;
+    }
+
+    return this.createNewEmptyWindow();
+  }
+
+  getEmptyWindowCount(): number {
+    let count = 0;
+
+    this._windows.forEach(sessionWindow => {
+      if (sessionWindow.contentViewType === ContentViewType.Welcome) {
+        count++;
+      }
+    });
+
+    return count;
+  }
+
+  private _getNewWindowRect(): IRect {
+    // cannot require the screen module until the app is ready, so require here
+    const { screen } = require('electron');
+    const primaryDisplay = screen.getPrimaryDisplay();
+    const {
+      width: screenWidth,
+      height: screenHeight
+    } = primaryDisplay.workAreaSize;
```

Comment:
So this does not work well for me with two-monitor setup. Say I have a setup like:

```
~~~~~
| 1  |  ~~~~~~~~~~~~
~~~~~   |          |
        |     2    |
        ~~~~~~~~~~~~
````

the window lands in the intersection of the bottom corner of 1 and top corner of 2. It is a bit impractical.

Reply:
[This comment](https://github.com/electron/electron/issues/3490) suggests using `.bounds`?

Reply:
There is also `.size` but neither `.bounds` nor `.workAreaSize` is well documented (at least not on https://www.electronjs.org/docs/latest/api/structures/display)

Reply:
can you test the latest? I updated with `bounds` origin.

Reply:
just tested and it worked better. pushed the change.

Reply:
This works well, cursor position is a neat heuristic. Thank you!